crates/matrix-sdk-base/src/event_cache_store/traits.rs

```diff
@@ -29,43 +29,81 @@ pub trait EventCacheStore: AsyncTraitDeps {
     /// The error type used by this event cache store.
     type Error: fmt::Debug + Into<EventCacheStoreError>;
 
-    /// Add a media file's content in the media store.
+    /// The retention policy set to cleanup the media cache.
+    async fn media_retention_policy(&self) -> Result<Option<MediaRetentionPolicy>, Self::Error>;
```

Comment:
This addition can go in a second commit:

- First commit: add `MediaRetentionPolicy`
- Second commit: use `MediaRetentionPolicy` along with tests.